### ISSUE TITLE
Migrate /pages/TestUserProfile.js to react-testing-library

### ISSUE THREAD

**Issue by bobsilverberg** [mozilla/addons-frontend#11304]
Migrate /pages/TestUserProfile.js to react-testing-library

**Comment by eviljeff** [comment]
as I'm planning to add a test to this file in https://github.com/mozilla/addons-frontend/issues/11315 I may as well give this a go at the same time

**Comment by bobsilverberg** [comment]
> as I'm planning to add a test to this file in #11315 I may as well give this a go at the same time

Amazing, go for it!

You might want to look at the PR I am about to put up as there is a small change to dealing with `history` in tests in it.

**Comment by eviljeff** [comment]
I think from the meeting you talked about changing our use of (or dropping) `sinon` too?  I can't recall what the alternative was if so.

**Comment by bobsilverberg** [comment]
> I think from the meeting you talked about changing our use of (or dropping) `sinon` too? I can't recall what the alternative was if so.

Yes, the alternative is to use `Jest`'s built-in mocking. There should be a few examples in the original PR for this project, as well as the one I just opened at https://github.com/mozilla/addons-frontend/pull/11318. Also, addons-code-manager uses Jest for mocking and it will have some examples. I think all of the code I have produced thus far just uses Jest to spy on `dispatch` and `history`, but we can use it for creating our own mocks too. If you run into anything that isn't obvious, feel free to ask about it.
